compiler/passes/src/destructuring/statement.rs

```diff
@@ -40,72 +40,83 @@ impl StatementReconstructor for DestructuringVisitor<'_> {
     fn reconstruct_assign(&mut self, mut assign: AssignStatement) -> (Statement, Self::AdditionalOutput) {
```

Comment:
`reconstruct_assign` has gotten fairly complex. Perhaps a detailed explanation of the expected transformation will improve clarity.